modules/javafx.graphics/src/main/java/com/sun/marlin/MaskMarlinAlphaConsumer.java

```diff
@@ -92,41 +92,35 @@ public MaskData getMaskData() {
 
     @Override
     public void setMaxAlpha(int maxalpha) {
-        ALPHA_MAP_USED = (maxalpha == 1) ? ALPHA_MAP_UNSAFE_NO_AA : ALPHA_MAP_UNSAFE;
+        ALPHA_MAP_USED = (maxalpha == 1) ? ALPHA_MAP_OFF_HEAP_NO_AA : ALPHA_MAP_OFF_HEAP;
     }
 
     // The alpha map used by this object (taken out of our map cache) to convert
     // pixel coverage counts (which are in the range [0, maxalpha])
     // into alpha values, which are in [0,255]).
-    static final byte[] ALPHA_MAP;
-    static final OffHeapArray ALPHA_MAP_UNSAFE;
+    static byte[] ALPHA_MAP;
+    static OffHeapArray ALPHA_MAP_OFF_HEAP;
 
-    static final byte[] ALPHA_MAP_NO_AA;
-    static final OffHeapArray ALPHA_MAP_UNSAFE_NO_AA;
-
-    static {
-        final Unsafe _unsafe = OffHeapArray.UNSAFE;
+    static byte[] ALPHA_MAP_NO_AA;
+    static OffHeapArray ALPHA_MAP_OFF_HEAP_NO_AA;
 
+    static void initAlphaMaps() {
         // AA:
         byte[] _ALPHA_MAP = buildAlphaMap(MarlinConst.MAX_AA_ALPHA);
         ALPHA_MAP = _ALPHA_MAP; // Keep alive the OffHeapArray
-        ALPHA_MAP_UNSAFE = new OffHeapArray(ALPHA_MAP, ALPHA_MAP.length); // 1K
-
-        long addr = ALPHA_MAP_UNSAFE.address;
+        ALPHA_MAP_OFF_HEAP = new OffHeapArray(ALPHA_MAP, ALPHA_MAP.length); // 1K
```

Comment:
L110: is `ALPHA_MAP` needed to keep the `parent` reference?  if so, why not use a single `Object` instead of an array?

Reply:
[JDK-8359260](https://bugs.openjdk.org/browse/JDK-8359260)